content/_code-samples/build-a-wallet/js/1_hello.js

```diff
@@ -0,0 +1,71 @@
+const { app, BrowserWindow } = require('electron')
+
+const path = require('path')
+const xrpl = require("xrpl")
+
+const testnetUrl = "wss://s.altnet.rippletest.net:51233"
+
+/**
+ * This function creates a WebService client, which connects to the XRPL and fetches the latest ledger index.
+ *
+ * @returns {Promise<number>}
+ */
+const getValidatedLedgerIndex = async () => {
+    const client = new xrpl.Client(testnetUrl)
+
+    await client.connect()
+
+    const ledgerRequest = {
+        "command": "ledger",
+        "ledger_index": "validated"
+    }
+
+    const ledgerResponse = client.request(ledgerRequest)
+
+    await client.disconnect()
+
+    return (await ledgerResponse).result.ledger_index
+}
+
+/**
+ * This is our main function, it creates our application window, preloads the code we will need to communicate
+ * between the renderer Process and the main Process, loads a layout and performs the main logic
+ */
+const createWindow = () => {
+
+    // Creates the application window
+    const appWindow = new BrowserWindow({
+        width: 800,
+        height: 600,
+        webPreferences: {
+            preload: path.join(__dirname, 'view', '1_preload.js'),
+        },
+    })
+
+    // Loads a layout
+    appWindow.loadFile(path.join(__dirname, 'view', '1_hello.html'))
+
+    // Performs main logic (display latest validated ledger index)
+    getValidatedLedgerIndex().then((value) => {
+        appWindow.webContents.send('update-ledger-index', value)
+    })
+
+}
+
+// For purposes of this tutorial we can safely ignore the following code
```

Comment:
Can you add a line explaining what it does at a high level, just so people modifying these snippets can get a sense for it, even if they don't modify it?

Reply:
This has been simplified, the comment has been removed